src/libs/actions/PersonalDetails.js

```diff
@@ -236,9 +235,10 @@ function mergeLocalPersonalDetails(details) {
  *
  * @param {Object} details
  * @param {boolean} shouldGrowl
+ * @returns {Promise}
  */
 function setPersonalDetails(details, shouldGrowl) {
-    API.PersonalDetails_Update({details: JSON.stringify(details)})
+    return API.PersonalDetails_Update({details: JSON.stringify(details)})
```

Comment:
Actions should never return a promise unless it is required in another action file. Seeing that this promise is used inside a view component is an anti-pattern.